app/gui2/shared/ast/tree.ts

```diff
@@ -34,12 +34,25 @@ import {
 declare const brandAstId: unique symbol
 export type AstId = string & { [brandAstId]: never }
 
+/** @internal */
+export type MetadataFields = {
+  externalId: ExternalId
+}
+export type NodeMetadataFields = {
+  position?: { x: number; y: number } | undefined
+  visualization?: VisualizationMetadata | undefined
+}
```

Comment:
I'm not sure. Although all nodes are `Ast`s, I wouldn't consider the `externalId` part of the node metadata. Their updates are currently handled separately, and in practice IDs have to be handled a bit specially--if an Ast has only one concrete child, that child should inherit the parent's `externalId` (this will be more important once we have `AssignmentExpression` nodes as a stable place to attach node metadata).

I'm not opposed to making `MetadataFields` the root of a metadata hierarchy if we find it would simplify anything, but currently (with the little metadata we have so far) I'm not seeing that to be the case.